dg-core/src/main/java/org/finra/datagenerator/engine/Frontier.java

```diff
@@ -26,13 +29,29 @@
 public interface Frontier {
 
     /**
+     * Generate Maps of variable assignments produced by a DFS strategy;
+     * the DFS is performed on a possible state bound up inside the Frontier and will
+     * stop early if flag is set to true
+     *
+     * @param singleThreadedProcessing Processing Strategy
+     * @param flag the exit flag
+     * @param outTemplate Output Template
+     * @param sb String Builder
+     * @return StringBuilder
+     * @throws IOException io exception
+     */
+    StringBuilder searchForScenarios(SingleThreadedProcessing singleThreadedProcessing, AtomicBoolean flag,
```

Comment:
Frontier needs to work without queues, with the processing strategy interface, and with no prescribed processing strategy interface. We can't hardcode SingleThreadedProcessing.